### PR TITLE
refactor(logging): separate logics for RDWS, ECS services and jobs.

### PR DESCRIPTION
**Pull request by Lou1415926**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/logging/workload.go [hunk 1]
```diff
@@ -31,200 +30,277 @@ type logGetter interface {
...
+		}
+		logEventsOpts.StreamLastEventTime = logEventsOutput.StreamLastEventTime
+		time.Sleep(cloudwatchlogs.SleepDuration)
+	}
 }
 
-func (o WriteLogEventsOpts) limit() *int64 {
-	if o.Limit != nil {
-		return o.Limit
+func (s *workloadLogger) ecsLogStreamPrefixes(taskIDs []string, container string) []string {
```
Comment (dannyrandall) [inline internal/pkg/logging/workload.go]: nit: i feel like this function doesn't belong on `workloadLogger` - maybe change the signature to something like
```
func ecsLogStreamPrefixes(service, container string, taskIDs []string) []string 
```

File: internal/pkg/logging/workload.go [hunk 2]
```diff
@@ -31,200 +30,277 @@ type logGetter interface {
...
 }
 
-func (o WriteLogEventsOpts) startTime(now func() time.Time) *int64 {
-	if o.StartTime != nil {
-		return o.StartTime
-	}
-	if o.Follow {
-		// Start following log events from current timestamp.
-		return aws.Int64(now().UnixMilli())
+// NewECSServiceClient returns an ECSServiceClient for the svc service under env and app.
```
Comment (dannyrandall) [inline internal/pkg/logging/workload.go]:
```suggestion
// NewECSServiceClient returns an ECSServiceClient for the service under env and app.
```

**Review by dannyrandall — APPROVED** [review]
lgtm, just a couple of small nits!

File: internal/pkg/logging/workload.go [hunk 3]
```diff
@@ -31,200 +30,277 @@ type logGetter interface {
...
+	return ecsLogStreamPrefixes(taskIDs, s.name, container)
+}
+
+// NewAppRunnerServiceLoggerOpts contains fields that initiate AppRunnerServiceLoggerOpts struct.
+type NewAppRunnerServiceLoggerOpts struct {
+	*NewWorkloadLoggerOpts
+	ConfigStore describe.ConfigStoreSvc
+}
 
+// NewAppRunnerServiceLogger returns an AppRunnerServiceLogger for the svc service under env and app.
```
Comment (paragbhingre) [inline internal/pkg/logging/workload.go]:
```suggestion
// NewAppRunnerServiceLogger returns an AppRunnerServiceLogger for the service under env and app.
```

File: internal/pkg/logging/workload.go [hunk 4]
```diff
@@ -31,200 +30,277 @@ type logGetter interface {
...
+
+// NewWorkloadLoggerOpts contains fields that initiate workloadLogger struct.
+type NewWorkloadLoggerOpts struct {
+	App  string
+	Env  string
+	Name string
+	Sess *session.Session
+}
+
+// newWorkloadLogger returns a workloadLogger for the svc service under env and app.
```
Comment (paragbhingre) [inline internal/pkg/logging/workload.go]:
```suggestion
// newWorkloadLogger returns a workloadLogger for the service under env and app.
```

File: internal/pkg/cli/svc_logs.go [hunk 5]
```diff
@@ -103,28 +114,29 @@ func newSvcLogOpts(vars wkldLogsVars) (*svcLogsOpts, error) {
...
 
-		if err != nil {
-			return err
+		newWorkloadLoggerOpts := &logging.NewWorkloadLoggerOpts{
+			App:  opts.appName,
+			Env:  opts.envName,
+			Name: opts.name,
+			Sess: sess,
+		}
+		switch opts.targetSvcType {
```
Comment (huanjani) [inline internal/pkg/cli/svc_logs.go]: nit: do we need a switch if there's only one case?

File: internal/pkg/logging/workload.go [hunk 6]
```diff
@@ -31,200 +30,277 @@ type logGetter interface {
...
+		if err != nil {
+			return fmt.Errorf("get log events for log group %s: %w", logEventsOpts.LogGroup, err)
+		}
+		if err := onEvent(s.w, cwEventsToHumanJSONStringers(logEventsOutput.Events)); err != nil {
+			return err
+		}
+		if !follow {
+			return nil
+		}
+		// for unit test.
```
Comment (huanjani) [inline internal/pkg/logging/workload.go]: tiny nit:
```suggestion
		// For unit test.
```

**Review by huanjani — APPROVED** [review]
Just nits! Looks good. Feel free to remove the label when ready!

**Pull request merged**